### PR TITLE
Feature #2583: Disable HTTP cloning

### PR DESCRIPTION
**Pull request by 0xbaadf00d**
Relative to #2583, this Pull Request allow administrator to disable any Git interactions by HTTP protocol.

```
[repository]
DISABLE_HTTP_GIT = true
```

image

image

### CODE REVIEW

File: modules/setting/setting.go [hunk 1]
```diff
@@ -116,6 +116,7 @@ var (
 		MirrorQueueLength      int
 		PullRequestQueueLength int
 		PreferredLicenses      []string
+		DisableHttpGit         bool
```
Comment (bkcsoft) [inline modules/setting/setting.go]: Change to `DisableHTTPGit`

File: routers/repo/http.go [hunk 2]
```diff
@@ -479,6 +479,11 @@ func HTTPBackend(ctx *context.Context, cfg *serviceConfig) http.HandlerFunc {
 		for _, route := range routes {
 			r.URL.Path = strings.ToLower(r.URL.Path) // blue: In case some repo name has upper case name
 			if m := route.reg.FindStringSubmatch(r.URL.Path); m != nil {
+				if setting.Repository.DisableHttpGit {
```
Comment (bkcsoft) [inline routers/repo/http.go]: Same as above (HTTP in upper-case)

File: templates/repo/wiki/view.tmpl [hunk 3]
```diff
@@ -29,15 +29,21 @@
...
-						{{if UseHTTPS}}HTTPS{{else}}HTTP{{end}}
-					</button>
+					{{if not $.DisableHTTP}}
+						<button class="ui basic clone button" id="repo-clone-https" data-link="{{.CloneLink.HTTPS}}">
+							{{if UseHTTPS}}HTTPS{{else}}HTTP{{end}}
+						</button>
+					{{end}}
 					{{if not $.DisableSSH}}
-						<button class="ui basic clone button" id="repo-clone-ssh" data-link="{{.WikiCloneLink.SSH}}">
+						<button class="ui basic clone button" id="repo-clone-ssh" data-link="{{.CloneLink.SSH}}">
```
Comment (bkcsoft) [inline templates/repo/wiki/view.tmpl]: Why change this but not remove `.WikiCloneLine` from the injected Data?

File: modules/setting/setting.go (hunk 1 above)
Comment (0xbaadf00d) [inline modules/setting/setting.go]: Done.

File: routers/repo/http.go (hunk 2 above)
Comment (0xbaadf00d) [inline routers/repo/http.go]: Done

**Comment by fboerman** [comment]
any news on when this is gonna be merged? really great feature!

**Comment by 0xbaadf00d** [comment]
I afraid it depend of @Unknwon. And for now, @Unknwon is not aware that he must take care of his project or nominate other maintainers to help him.

**Comment by fboerman** [comment]
yeah but he does commit sometimes :( well i applied it as git patch so that works as well

**Comment by 0xbaadf00d** [comment]
Maybe you will be more lucky than me to contact Unknwon on Gitter ou via discuss.gogs.io. To ask him to accept this PR.

**Comment by Unknwon** [comment]
Thanks!

**Pull request merged**